src/vcpkg/build.cpp

```diff
@@ -861,6 +864,34 @@ namespace vcpkg::Build
         }
     }
 
+    static void write_sbom(const VcpkgPaths& paths,
+                           const InstallPlanAction& action,
+                           std::vector<Json::Value> heuristic_resources)
+    {
+        auto& fs = paths.get_filesystem();
+        const auto& scfl = action.source_control_file_and_location.value_or_exit(VCPKG_LINE_INFO);
+        const auto& scf = *scfl.source_control_file;
+
+        auto doc_ns = Strings::concat("https://spdx.org/spdxdocs/",
+                                      scf.core_paragraph->name,
+                                      '-',
+                                      action.spec.triplet(),
+                                      '-',
+                                      scf.to_version(),
+                                      '-',
+                                      generate_random_UUID());
+
+        const auto now = CTime::get_current_date_time().value_or_exit(VCPKG_LINE_INFO).strftime("%Y-%m-%dT%H:%M:%SZ");
```

Comment:
I'm not sure what we can do to continue if getting the current datetime fails. We can't continue to create the binary package because it would result in different contents on different machines.

Looking at the ms docs[1], it seems that gmtime will fail only if passed a value before the unix epoch (i.e. a negative time):
>If sourceTime represents a date before midnight, January 1, 1970, gmtime returns NULL. There's no error return.

I don't know how much ceremony is warranted to guard against a user setting their system clock before Jan 1 1970.

[1]: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/gmtime-gmtime32-gmtime64?view=msvc-170

Reply:
yeah, fair enough; I don't love using `value_or_exit()`, but `get_current_date_time()` only fails when vcpkg is being used on a system from before the epoch, or on a 16-bit system from after the year 18000... so... really unlikely. I think it'd be best to have `get_current_date_time()` never fail, given this.